# Team1-ComparativeGenomics

**Team Members**: Laura Mora, Heather Patrick, Lawrence McKinney, Kenji Gerhardt, Hira Anis, Manasa Vegesna 

# Comparative Genomics Pipeline

This pipeline performs various comparative genomics techniques on the 50 E.coli isolates using ANI, MLST and SNP-based approaches to determine the outbreak cluster. 

## Tools Used:

## ANI - Approach
	
#### MUMmer:	
   * Description of the tool
   
## MLST Approach

#### ChewBBACA:	
   * Description of the tool
   
## SNP Approach

#### kSNP:
   * kSNP3 is a program that identifies the pan-genome SNPs in a set of genome sequences, and estimates
phylogenetic trees based upon those SNPs. 
   * Version: 3.0

### Usage:

### Requirememts:

### Citations:

1) Kurtz, S., Phillippy, A., Delcher, A. L., Smoot, M., Shumway, M., Antonescu, C., & Salzberg, S. L. (2004). Versatile and open software for comparing large genomes. Genome biology, 5(2), R12.
2) Silva, M., Machado, M. P., Silva, D. N., Rossi, M., Moran-Gilad, J., Santos, S., ... & Carrico, J. A. (2018). chewBBACA: A complete suite for gene-by-gene schema creation and strain identification. Microbial genomics, 4(3).
3) Gardner, S. N., Slezak, T., & Hall, B. G. (2015). kSNP3. 0: SNP detection and phylogenetic analysis of genomes without genome alignment or reference genome. Bioinformatics, 31(17), 2877-2878.
